.github/workflows/version-bump.yml

```diff
@@ -0,0 +1,93 @@
+# This workflow bumps the submodule version used for a particular project
+# In the case of Bonsai being updated, this also updates the installation page with the latest version info
+name: Update documented project version
+run-name: Update `${{github.event.inputs.project}}` to `${{github.event.inputs.version}}`
+on:
+  workflow_dispatch:
+    inputs:
+      project:
+        description: "The name of the project to be updated (IE: a folder name within the src directory)"
+        required: true
+      version:
+        description: "The target version to update to (IE: a Git tag in the project)"
+        required: true
+      project-fork-url:
+        description: "Git URL of the project for testing in forks"
+        default: ""
+concurrency:
+  group: version-bump
+permissions:
+  # Required to trigger GitHub Pages deployment
+  actions: write
+  # Required to push changes
+  contents: write
+jobs:
+  update:
+    name: Update ${{github.event.inputs.project}} to ${{github.event.inputs.version}}
+    runs-on: ubuntu-latest
+    env:
+      PROJECT: ${{github.event.inputs.project}}
+      VERSION: ${{github.event.inputs.version}}
+    steps:
+      # ----------------------------------------------------------------------- Checkout
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      # ----------------------------------------------------------------------- Override the submodule URL
+      # This is to support testing release automation in forks, it is disabled for the canonical docs repo
+      - name: Override ${{github.event.inputs.project}}'s submodule URL
+        if: vars.IS_CANONICAL_DOCS_REPO != 'true' && github.event.inputs.project-fork-url != ''
+        run: git config --local --add "submodule.src/$PROJECT.url" "$FORK_URL"
+        env:
+          FORK_URL: ${{github.event.inputs.project-fork-url}}
+
+      # ----------------------------------------------------------------------- Update the submodule
+      - name: Clone ${{github.event.inputs.project}} submodule
+        run: git submodule update --init "src/$PROJECT/"
+      - name: Update ${{github.event.inputs.project}} submodule
+        working-directory: src/${{github.event.inputs.project}}/
+        run: git checkout "refs/tags/$VERSION"
+      - name: Stage ${{github.event.inputs.project}} submodule
+        run: git add "src/$PROJECT/"
+
+      # ----------------------------------------------------------------------- Update the installation page
+      - name: Setup Python 3.10
```

Comment:
Any reason to prefer Python 3.10 here instead of 3.11 used in bonsai release automation? Just curious.

Reply:
TL;DR: They used to both be 3.10 but Bonsai needed a 3.11 feature. I'll make them both 3.11 for consistency.

------------------

It's probably easier to answer why Bonsai is different.

The reason I went with 3.10 to start is that it's the [default Python used by the runners](https://github.com/actions/runner-images/blob/0ad636b8243987e85a0148adcfd379b1661a0728/images/ubuntu/Ubuntu2204-Readme.md#language-and-runtime), which means the `setup-python` is a no-op today and future proofing tomorrow.

The reason the main repo is 3.11 is because I wanted `ZipFile.mkdir` for `create-portable-zip.py`, which was [added in 3.11](https://docs.python.org/3/whatsnew/3.11.html#zipfile)

I'll update this one to 3.11 just for the sake of consistency.

(In case you're wondering if that'll negatively impact the workflows, 3.11 is [in the runner tool cache](https://github.com/actions/runner-images/blob/0ad636b8243987e85a0148adcfd379b1661a0728/images/ubuntu/Ubuntu2204-Readme.md#python) so setup-python just switches to it without downloading or installing anything.)